mlx/primitives.h

```diff
@@ -314,6 +314,23 @@ class ArcTan : public UnaryPrimitive {
   void eval(const std::vector<array>& inputs, array& out);
 };
 
+class ArcTan2 : public UnaryPrimitive {
```

Comment:
Could you add it to this list as well: https://github.com/ml-explore/mlx/blob/main/mlx/compile.cpp#L38